internal/pkg/deploy/cloudformation/stack/transformers.go

```diff
@@ -210,6 +229,45 @@ func convertAutoscaling(a manifest.AdvancedCount) (*template.AutoscalingOpts, er
 		autoscalingOpts.Memory = aws.Float64(float64(*a.Memory.ScalingConfig.Value))
 	}
 
+	var (
+		CPUCooldown *template.Cooldown
+		MemCooldown *template.Cooldown
+	)
+
+	// Write from generalized cooldown first
+	if generalCooldown := convertCooldown(a.Cooldown); generalCooldown != nil {
+		CPUCooldown = generalCooldown
+		MemCooldown = generalCooldown
+	}
+
+	// If both generalized cooldown and specific cooldowns are specified, cooldown is overwritten by lower level config
+	if specCPUCooldown := convertCooldown(a.CPU.ScalingConfig.Cooldown); specCPUCooldown != nil {
+		if CPUCooldown == nil {
+			CPUCooldown = &template.Cooldown{}
+		}
+		if specCPUCooldown.ScaleInCooldown != nil {
+			CPUCooldown.ScaleInCooldown = specCPUCooldown.ScaleInCooldown
+		}
+		if specCPUCooldown.ScaleOutCooldown != nil {
+			CPUCooldown.ScaleOutCooldown = specCPUCooldown.ScaleOutCooldown
+		}
+	}
+
+	if specMemCooldown := convertCooldown(a.Memory.ScalingConfig.Cooldown); specMemCooldown != nil {
+		if MemCooldown == nil {
+			MemCooldown = &template.Cooldown{}
+		}
+		if specMemCooldown.ScaleInCooldown != nil {
+			MemCooldown.ScaleInCooldown = specMemCooldown.ScaleInCooldown
+		}
+		if specMemCooldown.ScaleOutCooldown != nil {
+			MemCooldown.ScaleOutCooldown = specMemCooldown.ScaleOutCooldown
+		}
+	}
+
+	autoscalingOpts.CPUCooldown = CPUCooldown
```

Comment:
Can we move the logic to a separate function to make this `convertAutoscaling` shorter?

Maybe 
```suggestion
	autoscalingOpts.CPUCooldown = convertScalingCooldown(a.CPU.ScalingConfig.Cooldown, generalCooldown)
```